foreman/client.py

```diff
@@ -383,30 +381,46 @@ def _get_local_defs(self, strict=True):
         :param strict: Use any version that shared major version and has lower
                        minor if no total match found
         """
+        version = parse_version(self.version)
         defs_path = os.path.join(os.path.dirname(__file__), 'definitions')
         files = glob.glob('%s/*-v%s.json' % (defs_path, self.api_version))
+        files = [
+            (fn, parse_version(os.path.basename(fn).split('-')[0]))
+            for fn in files
+        ]
+
         last_major_match = None
-        for f_name in sorted(files):
-            f_ver = os.path.basename(f_name).split('-')[0]
-            if f_ver == self.version:
+        last_lower_version = None
+        for f_name, f_ver in sorted(files, key=lambda x: x[1]):
```

Comment:
No, I'm wrong, that's what the zero-padding is for, but that will make the version 1dev to be considered greater than 2. That's not acceptable.

Is there any specific case why you want to support string in versions?